src/api.ts

```diff
@@ -0,0 +1,61 @@
+import { Api } from "coder/site/src/api/api"
+import fs from "fs/promises"
+import * as https from "https"
+import * as os from "os"
+import * as vscode from "vscode"
+import { CertificateError } from "./error"
+import { Storage } from "./storage"
+
+// expandPath will expand ${userHome} in the input string.
+const expandPath = (input: string): string => {
+  const userHome = os.homedir()
+  return input.replace(/\${userHome}/g, userHome)
+}
+
+/**
+ * Create an sdk instance using the provided URL and token and hook it up to
+ * configuration.  The token may be undefined if some other form of
+ * authentication is being used.
+ */
+export async function makeCoderSdk(baseUrl: string, token: string | undefined, storage: Storage): Promise<Api> {
+  const restClient = new Api()
+  restClient.setHost(baseUrl)
+  if (token) {
+    restClient.setSessionToken(token)
+  }
+
+  restClient.getAxiosInstance().interceptors.request.use(async (config) => {
```

Comment:
Not a huge deal, but is there a reason why the Axios instance isn't a separate variable?

Reply:
Though I guess another option would be to add both interceptors in one call of `interceptors.request.use`

Reply:
Oh like why not split out into `const axios = restClient.getAxiosInstance()` type of thing?

Could do that, I tend to like leaving things on their objects and accessing them that way, mostly just a preference thing I think.  I think maybe I like having them "namespaced" or something like that.

Reply:
Oh and one is a `request` interceptor and the other a `response` interceptor so I think they have to be separate calls

Reply:
Oh, wait – you're right. Totally missed that they were different names